RGLServerPlugin/include/RGLServerPluginInstance.hh

```diff
@@ -69,20 +69,43 @@ private:
                         bool paused);
     void RayTrace(std::chrono::steady_clock::duration sim_time);
 
-    ignition::msgs::PointCloudPacked CreatePointCloudMsg(std::chrono::steady_clock::duration sim_time, std::string frame, int hitpointCount);
-    ignition::msgs::LaserScan CreateLaserScanMsg(std::chrono::steady_clock::duration sim_time, std::string frame, int hitpointCount);
+    bool FetchLaserScanResult();
+    bool FetchPointCloudResult(rgl_node_t formatNode);
+
+    ignition::msgs::PointCloudPacked CreatePointCloudMsg(std::chrono::steady_clock::duration sim_time, const std::string& frame);
+    ignition::msgs::LaserScan CreateLaserScanMsg(std::chrono::steady_clock::duration sim_time, const std::string& frame);
 
     void DestroyLidar();
 
     std::string topicName;
     std::string frameId;
-    float lidarRange;
+    rgl_vec2f lidarMinMaxRange;
     ignition::math::Angle scanHMin;
     ignition::math::Angle scanHMax;
     int scanHSamples;
     std::vector<rgl_mat3x4f> lidarPattern;
-    std::vector<rgl_vec3f> resultPointCloud;
-    std::vector<float> resultDistances;
+
+    struct ResultPointCloud
+    {
+        std::vector<char> data{};
+        int32_t hitPointCount{};
+        const std::size_t pointSize{sizeof(rgl_vec3f) + sizeof(float)};  // Based on rglFields
+        const std::vector<rgl_field_t> rglFields = {
+                RGL_FIELD_XYZ_VEC3_F32,
+                RGL_FIELD_LASER_RETRO_F32
```

Comment:
I wanted to wrap up all the information about these results in one place. There will be no duplicates because only one object of these structures exists. I leave it as it is.